src/storage/core.rs

```diff
@@ -1212,3 +1297,37 @@ where
         active + closed
     }
 }
+
+/// Result of read operations
+#[derive(Debug)]
+pub enum ReadResult<T> {
+    /// Data was found
+    Found(T),
+    /// Data was deleted, contains creation timestamp
+    Deleted(u64),
+    /// Data was not found
+    NotFound,
+}
+
+impl<T> ReadResult<T> {
+    pub(crate) fn is_meaningful(&self) -> bool {
```

Comment:
IMHO, the function name is not good. What do you think about  `is_presented` name?

Also add more functions for simplicity: 
1. `is_found`
2. `is_deleted`
3. `is_not_found`